processor/metricsgenerationprocessor/README.md

```diff
@@ -0,0 +1,64 @@
+# Metrics Generation Processor
+**Status: under development; Not recommended for production usage.**
+
+Supported pipeline types: metrics
+
+## Description
+
+The metrics generation processor (`metricsgenerationprocessor`) can be used to create new metrics using existing metrics following a given rule. Currently it supports following two approaches for creating a new metric.
+
+1. It can create a new metric using two existing metrics applying one of the follwing arithmatic operations: add, subtract, multiply, divide and percent. One use case is to calculate the `pod.memory.utilization` metric like the following equation-
+`pod.memory.utilization` = (`pod.memory.usage.bytes` / `node.memory.limit`)
+1. It can create a new metric by scaling the value of an existing metric with a given constant number. One use case is to convert `pod.memory.usage` metric values from Megabytes to Bytes (multiply the existing metric's value by 1000000)
```

Comment:
Can any of the operations listed in the first capability statement be used with constant values? Would it make sense to?
```suggestion
1. It can create a new metric by scaling the value of an existing metric with a given constant number. One use case is to convert `pod.memory.usage` metric values from Megabytes to Bytes (multiply the existing metric's value by 1,048,576)
```

Reply:
> Can any of the operations listed in the first capability statement be used with constant values? Would it make sense to?

Yes, they can be. Two make the purpose clear I came up with the generationType idea where user has clear idea how they are calculating their metrics- is it only scaling a value or creating a metrics from two different existing metrics. We had a slight discussion in SIG meeting and many of us seems to be OK with this approach. Still, we can wait for more comments maybe.

Reply:
I think it definitely makes sense to be able to apply any of the operations with a fixed value.  My question here is mostly around the term "scaling" which implies multiplication/division.  Addition/subtraction would be offsetting rather than scaling.  I'm not sure what a better name would be that would capture both, though, so don't consider this a strong objection to "scaling".

Reply:
I see. I thought the term "scaling" makes sense for any operation with constant value. I also can't think of a more generic term here.